src/coredump/coredump.c

```diff
@@ -860,8 +863,9 @@ static int submit_coredump(
         JSON_VARIANT_OBJECT_FOREACH(module_name, module_json, json_metadata) {
                 JsonVariant *package_name, *package_version;
 
-                /* We only add structured fields for the 'main' ELF module */
-                if (!path_equal_filename(module_name, context->meta[META_EXE]))
+                /* We only add structured fields for the 'main' ELF module, and only if we can identify it. */
+                if (!context->meta[META_EXE] ||
```

Comment:
Can META_COMM be used here too?

Reply:
IIUC, no. When parsing the core file we use file names, and COMM could be anything.

Reply:
Is it worth trying at least some best-effort approach, eg `string_contains_word(module_name, context->meta[META_EXE])` that might give us some more hits?

Reply:
I don't think we care too much. The cases where META_EXE is not available should be fairly rare, only when something went wrong when reading the attributes. So I think it's fine to just skip this extra information in those cases.

Reply:
ok, in that case I'd do

```
if (!context->meta[META_EXE])
  break;
```

and avoid looping around

Reply:
Updated.